src/screens/AddNewItemScreen/index.tsx

```diff
@@ -1,25 +1,44 @@
 import React, { useState, FC } from 'react';
 import { Text, TextInput, TouchableOpacity, View } from 'react-native';
 import { style } from './style';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import { addItemAction } from '../../redux/actions/todoActions/addItemAction';
 import { COLORS } from '../../COLORS';
 import { createNewItemHelper } from '../../helpers/createNewItemHelper';
 import { TodoItemType } from '../../models';
 import { useNavigation } from '@react-navigation/native';
 import { AddNewItemScreenNavigationProps } from './type';
+import { firebase } from '@react-native-firebase/database';
+import { userTokenSelector } from '../../redux/selectors/userTokenSelector';
+import { SignInPayload } from '../../redux/actions/todoActions/signInAction';
 
 export const AddNewItemScreen: FC = () => {
   const navigation = useNavigation<AddNewItemScreenNavigationProps>();
-  const dispatch = useDispatch();
+  // const dispatch = useDispatch();
+
+  const userToken = useSelector(userTokenSelector);
 
   const [text, setText] = useState<string>('');
   const buttonStyle = text ? style.button : style.buttonDis;
 
-  const addItem = (text: TodoItemType['text']) => {
-    const newItem = createNewItemHelper(text);
+  const setData = async (
+    newItem: TodoItemType,
+    userToken: SignInPayload['userToken'],
+  ) => {
+    await firebase
```

Comment:
these lines duplicate every time when you are calling firebase 

```
firebase
      .app()
      .database(
        `https://fir-2f0d3-default-rtdb.europe-west1.firebasedatabase.app/`,
      )
```

make sense to move it to service helper